india_compliance/gst_india/utils/gstr/gstr.py

```diff
@@ -122,3 +123,64 @@ def set_key(self, key, value):
 
     def update_gstins(self):
         pass
+
+
+@frappe.whitelist()
+def validate_company_gstins(company=None, company_gstin=None):
+    """
+    Checks the validity of the company's GSTIN authentication.
+
+    Args:
+        company_gstin (str): The GSTIN of the company to validate.
+
+    Returns:
+        dict: A dictionary where the keys are the GSTINs and the values are booleans indicating whether the authentication is valid.
+    """
+    frappe.has_permission("GST Settings", throw=True)
+
+    credentials = get_company_gstin_credentials(company, company_gstin)
+
+    gstin_authentication_status = {
+        credential.gstin: (
+            credential.session_expiry
+            and credential.auth_token
+            and credential.session_expiry
+            > frappe.utils.now_datetime() + frappe.utils.datetime.timedelta(minutes=30)
+        )
+        for credential in credentials
+    }
+
+    return gstin_authentication_status
+
+
+def get_company_gstin_credentials(company=None, company_gstin=None):
+    filters = {"service": "Returns"}
+
+    if company:
+        filters["company"] = company
```

Comment:
Validate to ensure all company gstins are covered, if there are any missing credentials. You could use `get_gstin_list` for this.

Validations:
- All GSTINs for the company are set in credentials
- Or GSTIN Requested is set in credentials